cmd/dockerd/daemon.go

```diff
@@ -202,10 +202,14 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 	if err != nil {
 		return err
 	}
-	signal.Trap(func() {
+
+	err := signal.Trap(func() {
```

Comment:
Because the problem is the error reporting in the underlying goroutine, getting back an error and returning it won't be sufficient. Just pass in the logger as an `interface { Info(args...inteface{}) }` and call that.